conda/recipes/librmm/recipe.yaml

```
@@ -0,0 +1,106 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/prefix-dev/recipe-format/main/schema.json
+# Copyright (c) 2018-2025, NVIDIA CORPORATION.
+schema_version: 1
+
+context:
+  version: ${{ env.get("RAPIDS_PACKAGE_VERSION") }}
+  cuda_version: ${{ (env.get("RAPIDS_CUDA_VERSION") | split("."))[:2] | join(".") }}
+  cuda_major: ${{ (env.get("RAPIDS_CUDA_VERSION") | split("."))[0] }}
+  date_string: ${{ env.get("RAPIDS_DATE_STRING") }}
+  head_rev: ${{ git.head_rev( "." )[:8] }}
+
+recipe:
+  name: librmm-split
+
+cache:
+  source:
+    path: ../../..
+
+  requirements:
+    build:
+      - cmake ${{ cmake_version }}
+      - ninja
+      - ${{ compiler("c") }}
+      - ${{ compiler("cxx") }}
+      - ${{ compiler("cuda") }}
+      - cuda-version =${{ cuda_version }}
+      - ${{ stdlib("c") }}
+
+  build:
+    script:
+      file: build.sh
+      secrets:
+        - AWS_ACCESS_KEY_ID
+        - AWS_SECRET_ACCESS_KEY
+        - AWS_SESSION_TOKEN
+
+      env:
+        CMAKE_C_COMPILER_LAUNCHER: ${{ env.get("CMAKE_C_COMPILER_LAUNCHER", default="sccache") }}
+        CMAKE_CUDA_COMPILER_LAUNCHER: ${{ env.get("CMAKE_CUDA_COMPILER_LAUNCHER", default="sccache") }}
+        CMAKE_CXX_COMPILER_LAUNCHER: ${{ env.get("CMAKE_CXX_COMPILER_LAUNCHER", default="sccache") }}
+        CMAKE_GENERATOR: ${{ env.get("CMAKE_GENERATOR", default="Ninja") }}
+        PARALLEL_LEVEL: ${{ env.get("PARALLEL_LEVEL", default="4") }}
+        SCCACHE_BUCKET: ${{ env.get("SCCACHE_BUCKET", default="") }}
+        SCCACHE_IDLE_TIMEOUT: ${{ env.get("SCCACHE_IDLE_TIMEOUT", default="") }}
+        SCCACHE_REGION: ${{ env.get("SCCACHE_REGION", default="") }}
+        SCCACHE_S3_USE_SSL: ${{ env.get("SCCACHE_S3_USE_SSL", default="") }}
+        SCCACHE_S3_NO_CREDENTIALS: ${{ env.get("SCCACHE_S3_NO_CREDENTIALS", default="") }}
+        SCCACHE_S3_KEY_PREFIX: librmm-${{ env.get("RUNNER_ARCH", default="X64") | replace("X64", "linux64") | replace("ARM64", "aarch64") | lower }}
+
+outputs:
+  - package:
+      name: librmm
+      version: ${{ version }}
+    build:
+      string: cuda${{ cuda_major }}_${{ date_string }}_${{ head_rev }}
+      script: install_librmm.sh
+    requirements:
+      host:
+        - cmake ${{ cmake_version }}
+        - cuda-version =${{ cuda_version }}
+      run:
+        - ${{ pin_compatible("cuda-version", upper_bound="x", lower_bound="x") }}
+        - if: cuda_major == "11"
+          then: cudatoolkit
+        - fmt ${{ fmt_version }}
+        - spdlog ${{ spdlog_version }}
+      run_exports:
+        - ${{ pin_subpackage("librmm", upper_bound="x.x") }}
+      ignore_run_exports:
+        from_package:
+          - ${{ compiler("cuda") }}
+    tests:
+      - script:
+          - "test -d \"${PREFIX}/include/rmm\""
+    about:
+      homepage: ${{ load_from_file("python/librmm/pyproject.toml").project.urls.Homepage }}
+      license: ${{ load_from_file("python/librmm/pyproject.toml").project.license.text | replace(" ", "-") }}
```

We should probably change `python/librmm/pyproject.toml` to use `Apache-2.0` as an SPDX identifier. https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license

@vyasr Do you know why we chose to write it this way in #1529?

---

We should probably fix this across the board -- I would suggest raising this with ops (in case they know of restrictions that I do not know). If ops is supportive, let's open a build-planning issue and audit this for all repos. Non-OSS repos may need a different solution, but Apache/BSD-3 repos should be fixable.

---

Then this will look like:
```suggestion
      license: ${{ load_from_file("python/librmm/pyproject.toml").project.license }}
```

---

IIRC the current choices were made in order to guarantee compatibility with wheeltamer. @raydouglass may remember the exact list of "allowed" licenses. Given that we no longer run wheeltamer before releases that is a moot point, but I don't know if there are any subsequent scans that we have reinstated where this could still be a problem.

---

Good -- that's exactly what I wondered. If we can adopt a normal SPDX license identifier in our `pyproject.toml` files, we absolutely should. I am okay with that being a follow-up to this PR, though.

---

OK sweet then we can go ahead and synchronize the values in conda and pip so that we don't need any modification from one to the other.

---

Following up: I opened https://github.com/rapidsai/build-planning/issues/152 to propose switching to SPDX license expressions (like `license = "Apache-2.0"`) but they're not supported by setuptools yet.

To simplify things, we should update our `pyproject.toml` files to `license = { text = "Apache-2.0" }` as we roll out rattler-build. @jakirkham proposed that here: https://github.com/rapidsai/build-planning/issues/152#issuecomment-2651936521

We should avoid using `replace(" ", "-")` in the new recipes, just fix the source when needed.

---

Pushed up a commit adding the `-` in the `pyproject.toml` files and removing the `replace`